### PR TITLE
Color options for calendar and weather forecast

### PR DESCRIPTION
**Pull request by artifactdev**
> Calendar Module

* If set colored to 'true' there can be a color definded for each calendar

example config:

`config: {
                colored: true,
				calendars: [
                    {
                        url: 'http://www.calendarlabs.com/templates/ical/US-Holidays.ics',
            			symbol: 'calendar',
                        color: '#8cd7ac'
                    },
                    {
                        url: 'http://www.calendarlabs.com/templates/ical/US-Holidays.ics',
            			symbol: 'calendar',
                        color: '#da7e7e'
                    },
					
				],
			}`

calendar-module-colored

> Weather Forecast Module

* If set colored to 'true' min-temp get's a blue tone and max-temp a red-tone

example config:

`config: {
			// See 'Configuration options' for more information.
			location: 'Amsterdam,Netherlands',
			locationID: '', //Location ID from http://openweathermap.org/help/city_list.txt
			appid: 'abcde12345abcde12345abcde12345ab', //openweathermap.org API key.
            colored: true
		}`

weatherforecast-module-colored

### CODE REVIEW

**Comment by nhubbard** [comment]
@artifactdev I like this, but we need to fix the conflicts in the calendar and weatherforecast READMEs.

File: modules/default/calendar/calendar.js [hunk 1]
```diff
@@ -142,7 +148,13 @@ Module.register("calendar", {
 			}
 
 			titleWrapper.innerHTML = this.titleTransform(event.title) + repeatingCountTitle;
-			titleWrapper.className = "title bright";
+
```
Comment (roramirez) [inline modules/default/calendar/calendar.js]: This will not pass the Eslint check

File: modules/default/calendar/calendar.js [hunk 2]
```diff
@@ -323,6 +335,24 @@ Module.register("calendar", {
...
+	 */
+	colorForUrl: function (url) {
+		for (var c in this.config.calendars) {
+			var calendar = this.config.calendars[c];
+			if (calendar.url === url && typeof calendar.color === "string") {
+				return calendar.color;
+			}
+		}
+
+		return '#fff';
```
Comment (roramirez) [inline modules/default/calendar/calendar.js]: You need set double quotes

File: modules/default/weatherforecast/weatherforecast.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ Module.register("weatherforecast",{
 		lang: config.language,
 		fade: true,
 		fadePoint: 0.25, // Start on 1/4th of the list.
+        colored: false,
```
Comment (roramirez) [inline modules/default/weatherforecast/weatherforecast.js]: Set tab instead of spaces

File: modules/default/weatherforecast/weatherforecast.js [hunk 4]
```diff
@@ -120,6 +121,9 @@ Module.register("weatherforecast",{
 			var forecast = this.forecast[f];
 
 			var row = document.createElement("tr");
+            if (this.config.colored) {
```
Comment (roramirez) [inline modules/default/weatherforecast/weatherforecast.js]: Same problem here with spaces.

File: modules/default/calendar/calendar.js (hunk 1 above)
Comment (artifactdev) [inline modules/default/calendar/calendar.js]: Why? Whats the problem here?

**Comment by artifactdev** [comment]
fixed the reviewed issues

File: modules/default/calendar/README.md [hunk 5]
```diff
@@ -63,10 +65,48 @@ config: {
...
 
 #### Calendar configuration options:
-| Option                | Description
-| --------------------- | -----------
-| `url`	                | The url of the calendar .ical. This property is required. <br><br> **Possible values:** Any public accessble .ical calendar.
-| `symbol`              | The symbol to show in front of an event. This property is optional. <br><br> **Possible values:** See [Font Awesome](http://fontawesome.io/icons/) website.
-| `repeatingCountTitle`	| The count title for yearly repating events in this calendar.  <br><br> **Example:** `'Birthday'`
-| `user`                | The username for HTTP Basic authentication.
-| `pass`                | The password for HTTP Basic authentication.
+<table width="100%">
```
Comment (roramirez) [inline modules/default/calendar/README.md]: You should use the previus format. The change of READMEs  to Markdown format was done recently

File: modules/default/calendar/calendar.js (hunk 1 above)
Comment (roramirez) [inline modules/default/calendar/calendar.js]: Exactly is  nothing there. I'm commented there  just say there some problems.

**Comment by artifactdev** [comment]
@roramirez I think it should be fine now?

**Comment by XxXCobraXxX** [comment]
@artifactdev I'm new here and in the developer branche. My question is: how can i add this pull request to my MagicMirror? With a special Github command?

Thanks in advance

**Comment by artifactdev** [comment]
@XxXCobraXxX Since the pull request was approved it shouldn't take long time until the code hits the developer branch. I can't say which time it appears in the master branch. So I recommend to wait. Alternative you could checkout my Fork, where my enhancements are added.
https://github.com/artifactdev/MagicMirror

**Comment by XxXCobraXxX** [comment]
Ok thank you

**Comment by MichMich** [comment]
@artifactdev, Could you please add an updated CHANGLELOG to this PR? Thanks!

**Comment by artifactdev** [comment]
@MichMich Done.

**Comment by KX900** [comment]
@artifactdev 
the calendar color doesn't work at my mirror.
I have exactly the same like in your example.
It even shows me both calendars, but it stays white.
do you know why?
i just added:
colored:true,
and both calendars with a color.

**Comment by fewieden** [comment]
this is not merged into master yet, you have to use the develop branch to get it working

**Comment by KX900** [comment]
Oh okay,
didn't thought about that. Thank you!

**Pull request merged**